custom_model_runner/datarobot_drum/drum/drum.py

```diff
@@ -224,37 +225,49 @@ def run(self):
         if self.run_mode in [RunMode.SERVER, RunMode.SCORE]:
             self._run_fit_and_predictions_pipelines_in_mlpiper()
         elif self.run_mode == RunMode.FIT:
-            remove_temp_output = None
-            if not self.options.output:
-                self.options.output = mkdtemp()
-                remove_temp_output = self.options.output
-            self._run_fit_and_predictions_pipelines_in_mlpiper()
-            if self.options.output or not self.options.skip_predict:
-                create_custom_inference_model_folder(self.options.code_dir, self.options.output)
-            if not self.options.skip_predict:
-                self.run_test_predict()
-            if remove_temp_output:
-                print(
-                    "Validation Complete 🎉 Your model can be fit to your data, "
-                    "and predictions can be made on the fit model! \n"
-                    "You're ready to add it to DataRobot. "
-                )
-                shutil.rmtree(remove_temp_output)
-            else:
-                print("Success 🎉")
+            self.run_fit()
         elif self.run_mode == RunMode.PERF_TEST:
             CMRunTests(self.options, self.run_mode).performance_test()
         elif self.run_mode == RunMode.VALIDATION:
             CMRunTests(self.options, self.run_mode).validation_test()
         elif self.run_mode == RunMode.NEW:
             self._generate_template()
         elif self.run_mode == RunMode.PUSH:
+            options, run_mode, raw_arguments = setup_validation_options(self.options)
+            validation_runner = CMRunner(self.runtime)
+            validation_runner.options = options
+            validation_runner.run_mode = run_mode
+            validation_runner.raw_arguments = raw_arguments
+            validation_runner.run()
+            print(
+                "Your model was successfully validated locally! Now we will add it into DataRobot"
+            )
             drum_push(self.options)
         else:
             error_message = "{} mode is not implemented".format(self.run_mode)
             print(error_message)
             raise DrumCommonException(error_message)
 
+    def run_fit(self):
+        remove_temp_output = None
+        if not self.options.output:
+            self.options.output = mkdtemp()
+            remove_temp_output = self.options.output
+        self._run_fit_and_predictions_pipelines_in_mlpiper()
+        if self.options.output or not self.options.skip_predict:
+            create_custom_inference_model_folder(self.options.code_dir, self.options.output)
+        if not self.options.skip_predict:
+            self.run_test_predict()
+        if remove_temp_output:
```

Comment:
Maybe I am missing something.. (probably).. but why different outputs in this case?
Validation is done in both cases.. yet the message is different?

Reply:
If there is no output argument supplied, we assumethey're using fit for validation and give them a big success message.

Reply:
If there is an output we assume they want to use drum to produce a new model.